gr-iio/include/gnuradio/iio/device_sink.h

```diff
@@ -61,7 +71,9 @@ class IIO_API device_sink : virtual public gr::sync_block
                      const std::vector<std::string>& params,
                      unsigned int buffer_size = DEFAULT_BUFFER_SIZE,
                      unsigned int interpolation = 0,
-                     bool cyclic = false);
+                     bool cyclic = false,
```

Comment:
Wherever possible, use setters and minimize the number of arguments in the constructor. Here, for example, if tagged input is to be used, the developer needs to look up "cyclic" even if the default value is desired. Some setters can only be used right after initialization, so they don't get GRC callbacks. Maybe they log warnings if used while the flowgraph is running. That only applies to "cyclic" here, maybe others that already use setters.

The alternative is to use a param list of some sort, so that missing params are ok. Multiple constructor signatures work in some cases, but they can be confusing.

Reply:
So basically a one-time use setter? Wouldn't that be equally applicable to all those other parameters then?

Reply:
Yes. Having to specify interpolation because you want to use cyclic is equally confusing, and buffer_size more so.